packages/admin/cms-admin/src/generator/future/generateForm.ts

```diff
@@ -0,0 +1,281 @@
+    
+        const { data, error, loading, refetch } = useQuery<GQL${gqlType}Query, GQL${gqlType}QueryVariables>(
+            ${instanceGqlType}Query,
+            id ? { variables: { id } } : { skip: true },
+        );
+    
+        const initialValues = React.useMemo<Partial<FormValues>>(() => data?.${instanceGqlType}
+        ? {
+            ...filter<GQL${fragmentName}Fragment>(${instanceGqlType}FormFragment, data.${instanceGqlType}),
+            ${numberFields.map((field) => `${String(field.name)}: String(data.${instanceGqlType}.${String(field.name)}),`).join("\n")}
+            ${Object.keys(rootBlocks)
+                .map((rootBlockKey) => `${rootBlockKey}: rootBlocks.${rootBlockKey}.input2State(data.${instanceGqlType}.${rootBlockKey}),`)
+                .join("\n")}
+        }
+        : {
+            ${booleanFields.map((field) => `${String(field.name)}: false,`).join("\n")}
+            ${Object.keys(rootBlocks)
+                .map((rootBlockKey) => `${rootBlockKey}: rootBlocks.${rootBlockKey}.defaultValues(),`)
+                .join("\n")}
+        }
+    , [data]);
+    
+        const saveConflict = useFormSaveConflict({
+            checkConflict: async () => {
+                const updatedAt = await queryUpdatedAt(client, "${instanceGqlType}", id);
+                return resolveHasSaveConflict(data?.${instanceGqlType}.updatedAt, updatedAt);
+            },
+            formApiRef,
+            loadLatestVersion: async () => {
+                await refetch();
+            },
+        });
+    
+        const handleSubmit = async (formValues: FormValues, form: FormApi<FormValues>, event: FinalFormSubmitEvent) => {
+            if (await saveConflict.checkForConflicts()) throw new Error("Conflicts detected");
+            const output = {
+                ...formValues,
+                ${numberFields.map((field) => `${String(field.name)}: parseFloat(formValues.${String(field.name)}),`).join("\n")}
+                ${Object.keys(rootBlocks)
+                    .map((rootBlockKey) => `${rootBlockKey}: rootBlocks.${rootBlockKey}.state2Output(formValues.${rootBlockKey}),`)
+                    .join("\n")}
+            };
+            if (mode === "edit") {
+                if (!id) throw new Error();
+                await client.mutate<GQLUpdate${gqlType}Mutation, GQLUpdate${gqlType}MutationVariables>({
+                    mutation: update${gqlType}Mutation,
+                    variables: { id, input: output, lastUpdatedAt: data?.${instanceGqlType}.updatedAt },
+                });
+            } else {
+                const { data: mutationReponse } = await client.mutate<GQLCreate${gqlType}Mutation, GQLCreate${gqlType}MutationVariables>({
+                    mutation: create${gqlType}Mutation,
+                    variables: { input: output },
+                });
+                if (!event.navigatingBack) {
+                    const id = mutationReponse?.create${gqlType}.id;
```

Comment:
NIT: stumbled upon this typo

```suggestion
                const { data: mutationResponse } = await client.mutate<GQLCreate${gqlType}Mutation, GQLCreate${gqlType}MutationVariables>({
                    mutation: create${gqlType}Mutation,
                    variables: { input: output },
                });
                if (!event.navigatingBack) {
                    const id = mutationResponse?.create${gqlType}.id;
```

Reply:
IMO this is not a nit, since this generated code is best practice 😁